src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensRefreshListener.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens
+{
+    /// <summary>
+    /// Sends a notification from server->client indicating something has changed in the LSP workspace.
+    /// The client will then send a request to the server for refreshed tokens.
+    /// </summary>
+    internal class SemanticTokensRefreshListener : IDisposable
+    {
+        private readonly LspWorkspaceManager _lspWorkspaceManager;
+        private readonly JsonRpc _jsonRpc;
+        private readonly AsyncBatchingWorkQueue _workQueue;
+
+        public SemanticTokensRefreshListener(
+            LspWorkspaceManager lspWorkspaceManager,
+            JsonRpc jsonRpc,
```

Comment:
It might be useful to implement an interface around this that basically has one method e.g. `SendNotification...`.  That way we aren't exposing the jsonrpc object directly to anyone (in case we ever change serialization methods etc.